server/jetstream_api.go

```diff
@@ -1825,17 +1829,45 @@ func (s *Server) jsStreamInfoRequest(sub *subscription, c *client, a *Account, s
 	// Check if they have asked for subject details.
 	if subjects != _EMPTY_ {
 		if mss := mset.store.SubjectsState(subjects); len(mss) > 0 {
-			if len(mss) > JSMaxSubjectDetails {
-				resp.StreamInfo = nil
-				resp.Error = NewJSStreamInfoMaxSubjectsError()
-				s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
-				return
+			// As go iterates over map in a non-consistent order, no choice but to buffer it a slice
+			type SubjectsState struct {
+				subject  string
+				messages uint64
 			}
-			sd := make(map[string]uint64, len(mss))
+
+			buffer := make([]SubjectsState, len(mss))
+			var i int
 			for subj, ss := range mss {
-				sd[subj] = ss.Msgs
+				buffer[i] = SubjectsState{subj, ss.Msgs}
```

Comment:
Just use append and add 0 to make above.

Reply:
Then don't have to do the `i` thing.

Reply:
ok